test: Added a mock server for openai testing

jsumners-nr opened the pull request:
This PR adds a mock OpenAI server utilizing the mock responses in the [python agent](https://github.com/newrelic/newrelic-python-agent/blob/02ad97def2808921f102dd7dc1b5e381d4ddaa7e/tests/mlmodel_openai/_mock_external_openai_server.py#L30-L166). This resolves #1846.

bizob2828 commented on test/lib/openai-mock-server.js:
```diff
@@ -0,0 +1,234 @@
+
+module.exports = openaiMockServer
+
+const RESPONSES = {
```
might be good to move the responses to its own file

bizob2828 reviewed (approved):
overall this is a great start. i can see us adding more mock responses that a have full list of headers/response keys to assert full event types. Aside from moving the mocks to its own file this works as expected.

jsumners-nr commented:
@bizob2828 I have refactored this PR to use a responses file in commit https://github.com/newrelic/node-newrelic/pull/1858/commits/af78c05d2ca89fc60933325bc11bdd761db95997. I opted to use a `Map` because:

1. It's easier to determine if a response exists (`.has`)
2. I find it easier to read individual responses when they are not in one massive object literal block

Note: I also fixed a mistake in the `.eslintrc.js` file.

bizob2828 commented on .eslintrc.js:
```diff
@@ -26,9 +26,6 @@ module.exports = {
       }
     ]
   },
-  parserOptions: {
```
oops, nice catch
